csv_generator/consumers/version_consumer.py

```diff
@@ -0,0 +1,37 @@
+from bs4 import BeautifulSoup
+
+from consumers.manuscript_consumer import ManuscriptXMLConsumer
+from consumers.utils import convert_ms_type
+
+
+class VersionXMLConsumer(ManuscriptXMLConsumer):
+    base_file_name = 'versions.csv'
+    headers = ['create_date',
+               'zip_name',
+               'xml_file_name',
+               'msid',
+               'version_position_in_ms',
+               'decision',
+               'ms_type']
+
+    @staticmethod
+    def get_versions(soup: BeautifulSoup) -> str:
+        return soup.find_all('version')
+
+    def process(self, soup: BeautifulSoup, xml_file_name: str) -> None:
+        """Parse target`BeautifulSoup` object, extract required data and
+        write data row to output_file.
+
+        :param soup:
+        :param xml_file_name:
+        :return:
+        """
+        msid = self.get_msid(soup)
+
+        versions = self.get_versions(soup)
+
+        for index, version in enumerate(versions):
+            decision = self.get_contents(version, 'decision')
+            ms_type = self.get_contents(version, 'manuscript-type')
```

Comment:
I didn't quite realise we have _manuscript-type_ in the XML in which case I am not sure whether we need that mapping (I thought we had to map from the code to the long type).

Reply:
Actually that shouldn't have surprised me since I am importing and using it in _PeerScout_